allow for git commit hashes to be passed to git open --commit command

wingleung opened the pull request:
build on top of the work done in #125 , allowing commit hashes to be passed as a specific commit instead of the current commit.

ffes commented:
Code looks good to me, but there is no mention if this in the man page, see `git-open.1.md` for that. And there are no tests. Please add them as well.

wingleung commented:
Updated with test and updated man entry.

But the code might be a little too hacky, wdyt?
Problem is I didn't find a way of allowing options like --commit to have a value optionally.
I didn't find much info about `OPTIONS_SPEC` on how to form the wildcards on the command options so I just picked the 3rd option as the hash is it exists.

The downside is that `git open --commit=some_commit_hash` won't work.
Only `git open --commit` and `git open --commit some_commit_hash` wil work.

Does anyone know how to have optional values on command options?

derimagia commented:
`git open origin <hash>` already works for this, is that not an option for you?

wingleung commented:
@derimagia for me it's not an issue but for people who use the equal sign in their options it won't work.

For example, the already existing suffix flag does work with the equal sign `git open --suffix=some_suffix`

derimagia commented:
`git open -c origin some_commit_hash` is the intended way for this to work. `some_commit_hash` is not an value for the `--commit` option but an argument to git-open.  git-open accepts a `ref` for the first parameter which can be a branch, commit hash, etc.

I would vote for an option to detect the remote (`origin`) - Ideally without a flag but I can't see a way around one at the moment

derimagia commented:
Going to close this. Also #166 fixes the equal signs issue mentioned
